src/main/java/xyz/nucleoid/plasmid/game/common/team/TeamManager.java

```diff
@@ -0,0 +1,330 @@
+package xyz.nucleoid.plasmid.game.common.team;
+
+import it.unimi.dsi.fastutil.objects.Object2BooleanArrayMap;
+import it.unimi.dsi.fastutil.objects.Object2BooleanMap;
+import it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.network.Packet;
+import net.minecraft.network.packet.s2c.play.TeamS2CPacket;
+import net.minecraft.scoreboard.AbstractTeam;
+import net.minecraft.scoreboard.Scoreboard;
+import net.minecraft.scoreboard.Team;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.text.LiteralText;
+import net.minecraft.text.Style;
+import net.minecraft.text.Text;
+import net.minecraft.util.ActionResult;
+import org.jetbrains.annotations.Nullable;
+import xyz.nucleoid.plasmid.game.GameActivity;
+import xyz.nucleoid.plasmid.game.event.GamePlayerEvents;
+import xyz.nucleoid.plasmid.game.player.MutablePlayerSet;
+import xyz.nucleoid.plasmid.game.player.PlayerSet;
+import xyz.nucleoid.plasmid.util.PlayerRef;
+import xyz.nucleoid.stimuli.event.player.PlayerDamageEvent;
+
+import java.util.*;
+
+/**
+ * Simple, GameActivity specific team manager class
+ */
+public final class TeamManager {
+    private final Map<GameTeam, Set<PlayerRef>> teamToPlayerRefSet = new Object2ObjectOpenHashMap<>();
+    private final Map<GameTeam, MutablePlayerSet> teamToPlayerSet = new Object2ObjectOpenHashMap<>();
+    private final Map<PlayerRef, GameTeam> playerRefToTeam = new Object2ObjectOpenHashMap<>();
+    private final Map<GameTeam, Team> teamToVanillaTeam = new Object2ObjectOpenHashMap<>();
+    private final Object2BooleanMap<GameTeam> teamFriendlyFire = new Object2BooleanArrayMap<>();
+
+    private final Scoreboard scoreboard = new Scoreboard();
+    private final MinecraftServer server;
+    private GameActivity gameActivity;
```

Comment:
The mutable state for the activity isn't ideal - is it needed? Could we avoid it by having `applyTo` return an initialized instance of a `TeamManager`?

Reply:
I could switch it technically for GameSpace, hovewer I prefer to keep it mutable so TeamManager can be passed to another GameActivity

Reply:
Hmm, I believe it would not really be possible to pass between activities anyway given the player remove events and close events will be called. This is not really something that can be worked around at the moment, at least until we do the deeper refactors.

Reply:
I think it will work fine, as long as it's reapplied